src/olympia/promoted/models.py

```diff
@@ -122,6 +222,75 @@ def save(self, *args, **kwargs):
             )
 
 
+# TODO: Drop Promotion suffix after dropping PromotedAddon table
+class PromotedAddonPromotion(ModelBase):
+    promoted_group = models.ForeignKey(
+        PromotedGroup,
+        on_delete=models.CASCADE,
+        null=False,
+        help_text='Can be set to Not Promoted to disable promotion without '
+        'deleting it.  Note: changing the group does *not* change '
+        'approvals of versions.',
+    )
+    addon = models.ForeignKey(
+        Addon,
+        on_delete=models.CASCADE,
+        null=False,
+        help_text='Add-on id this item will point to (If you do not know the '
+        'id, paste the slug instead and it will be transformed '
+        'automatically for you. If you have access to the add-on '
+        'admin page, you can use the magnifying glass to see '
+        'all available add-ons.',
+    )
+    application_id = models.SmallIntegerField(
+        choices=APPS_CHOICES,
+        null=False,
```

Comment:
The current model supports nulls, and use that to mean "all applications". Are you intending to support that through `0` ? If so, that's not documented, and not part of `APPS_CHOICES`.

Reply:
That is supported by virtue of the fact that we create a unique instance of `PromotedAddonPromotion` per application. so we do not need to support null or 0 as both are defined by the data model

Null -> simply no instances
0 -> 2 instances

That difference is one of the primary reasons for creating a new model rather than trying to roll this change into the existing one.